src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
From what I know of the wayland protocol, each seat has its own pointer/keyboard focus.

Also, as the compositor forwards raw key events to the clients and they are supposed to handle the keymap themselves, via libxkbcommon. From a protocol point of view, switching keyboard layout implies sending a new `wl_keyboard::keymap` event to all clients, and these clients are probably supposed to load a new keymap themselves. So, merging two keyboards with different layouts sounds complicated...

However, I think that properly handling runtime switch of keyboard layout is a feature we want to have, and the helpers smithay will provide to handle the `wl_seat` should provide it.